dev/com.ibm.ws.logging_2_fat/fat/src/com/ibm/ws/logging/fat/MessageTraceFileNameTimedRolloverTest.java

```diff
@@ -0,0 +1,273 @@
+        server.waitForStringInLog("CWWKF0011I");
+
+        // Dynamically update server.xml to change traceFileName to test.log
+        server.setMarkToEndOfLog();
+        server.setServerConfigurationFile("server_test_log.xml");
+
+        // Wait for config update
+        assertNotNull("Configuration update should complete",
+                      server.waitForConfigUpdateInLogUsingMark(null));
+
+        // Verify new test.log gets created
+        boolean testLogNowExists = server.fileExistsInLibertyServerRoot("logs/test.log");
+        assertTrue("REQUIREMENT: test.log should exist after dynamic update to traceFileName=test.log", testLogNowExists);
+    }
+
+    /**
+     * Test 5: Repeat scenario 4 using messageFileName, instead.
+     * Configure TimedLogRollover (in server.env), then dynamically update messageFileName
+     * to custom file, verify rollover behavior.
+     */
+    @Test
+    public void testDynamicMessageFileNameToCustomFile() throws Exception {
+        // Start with default configuration
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        // Dynamically update server.xml to change messageFileName to custom_message.log
+        server.setMarkToEndOfLog();
+        server.setServerConfigurationFile("server_custom_message.xml");
+
+        // Wait for config update
+        server.waitForStringInLogUsingMark("CWWKG0017I");
+
+        // Verify new custom_message.log gets created
+        boolean customMessageNowExists = server.fileExistsInLibertyServerRoot("logs/custom_message.log");
+        assertTrue("REQUIREMENT: custom_message.log should exist after dynamic update to messageFileName=custom_message.log", customMessageNowExists);
+    }
+
+    /**
+     * Test 6: Configure traceFileName=stdout in server.xml, verify the trace.log is NOT created.
+     * (This scenario depends on issue #31949 - Cannot stop generating trace.log file)
+     */
+    @Test
+    public void testTraceFileNameStdoutNoTraceLogCreated() throws Exception {
+        // Use server.xml configuration with traceFileName=stdout (same as Test 2 approach)
+        server.setServerConfigurationFile("server_trace_stdout.xml");
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        // Main test: Verify trace.log is NOT created when traceFileName=stdout in server.xml
+        boolean traceLogExists = server.fileExistsInLibertyServerRoot("logs/trace.log");
+        assertFalse("REQUIREMENT: trace.log should NOT be created when traceFileName=stdout in server.xml", traceLogExists);
+
+        // Verify other logs are still functional
+        boolean messageLogExists = server.fileExistsInLibertyServerRoot("logs/messages.log") ||
+                                   server.fileExistsInLibertyServerRoot("logs/message.log");
+        assertTrue("Other log files should still be created normally", messageLogExists);
+    }
+
+    // Helper methods (copied from the example file)
```

Comment:
We already have some existing similar helper methods in LibertyServer.java, please take a look and use as appropriate for the following helper methods.